extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/TrafficShapingConfig.java

```diff
@@ -0,0 +1,77 @@
+package io.quarkus.vertx.http.runtime;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.configuration.MemorySize;
+
+/**
+ * Configure the global traffic shaping functionality.
+ * It allows you to limit the bandwidth across all channels, regardless of the number of open channels.
+ * This can be useful when you want to control the overall network traffic to prevent congestion
+ * or prioritize certain types of traffic.
+ * <p>
+ * The traffic shaping allows you to configure various parameters, such as write and read limitations (in bytes per
+ * second), check interval (the delay between two computations of the bandwidth), and maximum time to wait.
+ * The check interval represents the period at which the traffic is computed, and a higher interval may result in
+ * less precise traffic shaping. It is recommended to set a positive value for the check interval, even if it is high,
+ * to ensure traffic shaping without accounting. A suggested value is something close to 5 or 10 minutes.
+ * <p>
+ * The `outbound-global-bandwidth` and `inbound-global-bandwidth` parameters represent the maximum number of bytes per second
+ * for write and read operations, respectively.
+ * Additionally, you can set the maximum time to wait, which specifies an upper bound for time shaping.
+ * By default, it is set to 15 seconds.
```

Comment:
Is this correct? Where does this show up in the config itself?

Reply:
That default is in Netty. 
See https://github.com/wuyinxian124/netty4.0.27Learn/blob/master/handler/src/main/java/io/netty/handler/traffic/AbstractTrafficShapingHandler.java#L58.